### PR TITLE
RNN 10.7 proofread full

### PR DESCRIPTION
**Pull request by goldmermaid**
Final 4 lines need to be reviewed.

### CODE REVIEW

File: chapter_recurrent-neural-networks/bptt.md [hunk 1]
```diff
@@ -140,22 +154,26 @@ $$\begin{aligned}
...
+1. Prove that for a random vector $\mathbf{x} \in \mathbb{R}^n$ with high probability $\mathbf{M}^k$, $\mathbf{x}$ will be very much likely aligned with the largest eigenvector $\mathbf{v}_n$ of $\mathbf{M}$. Formalize this statement.
 1. What does the above result mean for gradients in a recurrent neural network?
-1. Besides gradient clipping, can you think of any other methods to cope with gradient explosion in recurrent neural networks?
+1. Besides gradient clipping, can you think of any other methods to cope with gradient explosion of recurrent neural networks?
 
 ## Scan the QR Code to [Discuss](https://discuss.mxnet.io/t/2366)
 
 ![](../img/qr_bptt.svg)
+
+Need to review:
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/bptt.md]: What's this?

Comment (goldmermaid) [reply chapter_recurrent-neural-networks/bptt.md]: The math that we need to discuss about when you are free.

Comment (astonzhang) [reply chapter_recurrent-neural-networks/bptt.md]: Then this section is not ready for merging. Can you add [WIP] in the title?

File: chapter_recurrent-neural-networks/bptt.md [hunk 2]
```diff
@@ -140,21 +154,21 @@ $$\begin{aligned}
...
+* For efficient computation, intermediate values are cached.
 
 ## Exercises
 
-1. Assume that we have a symmetric matrix $\mathbf{M} \in \mathbb{R}^{n \times n}$ with eigenvalues $\lambda_i$. Without loss of generality assume that they are ordered in ascending order $\lambda_i \leq \lambda_{i+1}$. Show that $\mathbf{M}^k$ has eigenvalues $\lambda_i^k$.
-1. Prove that for a random vector $\mathbf{x} \in \mathbb{R}^n$ with high probability $\mathbf{M}^k \mathbf{x}$ will by very much aligned with the largest eigenvector $\mathbf{v}_n$ of $\mathbf{M}$. Formalize this statement.
+1. Assume that we have a symmetric matrix $\mathbf{M} \in \mathbb{R}^{n \times n}$ with eigenvalues $\lambda_i$. Without loss of generality, assume that they are ordered in an ascending order $\lambda_i \leq \lambda_{i+1}$. Show that $\mathbf{M}^k$ has eigenvalues $\lambda_i^k$.
+1. Prove that for a random vector $\mathbf{x} \in \mathbb{R}^n$ with high probability $\mathbf{M}^k$, $\mathbf{x}$ will be very much likely aligned with the largest eigenvector $\mathbf{v}_n$ of $\mathbf{M}$. Formalize this statement.
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/bptt.md]: for XXX, with high probability YYY will be .........

File: chapter_recurrent-neural-networks/bptt.md [hunk 3]
```diff
@@ -8,125 +8,139 @@ section we invoked `s.detach()` on the sequence. None of this was really fully
...
-how) the math works. For a more detailed discussion, e.g., about
-randomization and backprop also see the paper by
+how) the math works. For a more detailed discussion about
+randomization and backpropagation also see the paper by
 [Tallec and Ollivier, 2017](https://arxiv.org/abs/1705.08209).
 
 We encountered some of the effects of gradient explosion when we first
 implemented recurrent neural networks (:numref:`sec_rnn_scratch`). In
-particular, if you solved the problems in the problem set, you would
+particular, if we solved the problems in the problem set, you would
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/bptt.md]: why "we"?

Comment (goldmermaid) [reply chapter_recurrent-neural-networks/bptt.md]: Sounds good

File: chapter_recurrent-neural-networks/bptt.md (hunk 2 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/bptt.md]: My fault

**Pull request merged**